classes/ValidFormBuilder/MultiField.php

```diff
@@ -147,6 +148,8 @@ public function __construct($label, $meta = array())
 
         $this->__dynamic = $this->getMeta("dynamic", $this->__dynamic);
         $this->__dynamicLabel = $this->getMeta("dynamicLabel", $this->__dynamicLabel);
+
+        $this->initialiseDynamicRemoveMeta();
```

Comment:
This is very bad practice when it comes to traits. You are assuming that this class has the trait `CanRemoveDynamicFields` by calling the method `initialiseDynamicRemoveMeta`. A trait should be something you can add and remove without changing something in the implementing class. And no, a check if the method exists is not a solution.

You could add an interface that forces the creation of the needed method in the implementing class, but then you would need to create a stub method. Not really pretty.

I think we need to have a better look at the architecture. The idea, I think, is that you want to add extra functionality to one or more classes. A correct implementation using traits is to extend existing methods and add the functionality there.

Looks like you are hooking into `__construct`, `addField` and `__toHtml`. These methods should be extended by the trait. Unfortunately the code base is still not segmented enough to really implement this effectively. Time for some refactoring...